sbncode/CAFMaker/FillTrue.cxx

```diff
@@ -127,6 +130,190 @@ namespace caf {
+            closest_dist = (particle.Position(i_traj).Vect() - loc_nosce_v).Mag();
+            traj_index = i_traj;
+          }
+        }
+
+        // residual range
+        truep.rr = 0;
+        if (traj_index >= 0) {
+          for (int i_traj = traj_index+1; i_traj < (int)particle.NumberTrajectoryPoints(); i_traj++) {
+            truep.rr += (particle.Position(i_traj).Vect() - particle.Position(i_traj-1).Vect()).Mag();
+          }
+
+          // Also account for the distance from the Hit point to the matched trajectory point
+          double hit_distance_along_particle = (loc_nosce_v - particle.Position(traj_index).Vect()).Dot(particle.Momentum(traj_index).Vect().Unit());
+          truep.rr += -hit_distance_along_particle;
+        }
+        else truep.rr = -1;
+
+        // pitch
+        //
+        // The true pitch is affected by the presence of space charge, so we cannot
+        // directly apply the true direction. We include the effect of space charge
+        // on the true pitch here
+        if (closest_dist >= 0. && direction.Mag() > 1e-4) {
+          geo::PlaneID plane(srtrack.producer, p.tpc, iplane);
+          float angletovert = geo->WireAngleToVertical(geo->View(plane), plane) - 0.5*::util::pi<>();
+          TVector3 loc_mdx_v = loc_nosce_v - direction * (geo->WirePitch(geo->View(plane)) / 2.);
+          TVector3 loc_pdx_v = loc_nosce_v + direction * (geo->WirePitch(geo->View(plane)) / 2.);
+
+          // Convert types for helper functions
+          geo::Point_t loc_mdx(loc_mdx_v.X(), loc_mdx_v.Y(), loc_mdx_v.Z());
+          geo::Point_t loc_pdx(loc_pdx_v.X(), loc_pdx_v.Y(), loc_pdx_v.Z());
+
+          // Map to wires
+          if (sce && sce->EnableSimSpatialSCE()) { 
+            int corr = geo->TPC(plane.TPC).DriftDir()[0];
+
+            geo::Vector_t offset_m = sce->GetPosOffsets(loc_mdx);
+            offset_m.SetX(offset_m.X()*corr); // convert from drift direction to detector direction
+            loc_mdx = loc_mdx + offset_m;
+
+            geo::Vector_t offset_p = sce->GetPosOffsets(loc_pdx);
+            offset_p.SetX(offset_p.X()*corr); // convert from drift direction to detector direction
+            loc_pdx = loc_pdx + offset_p;
+          }
+
+          // Direction at wires
+          geo::Vector_t dir = (loc_pdx - loc_mdx) /  (loc_mdx - loc_pdx).r();
+
+          // Pitch at wires
+          double cosgamma = std::abs(std::sin(angletovert)*dir.Y() + std::cos(angletovert)*dir.Z());
+          double pitch;
+          if (cosgamma) {
+            pitch = geo->WirePitch(geo->View(plane))/cosgamma;
+          }
+          else {
+            pitch = 0.;
+          }
+
+          // Bring it back to particle trajectory
+          geo::Point_t loc_atwires_alongpitch = loc_sce + dir*pitch; 
+          geo::Point_t loc_alongpitch;
+          if (sce && sce->EnableSimSpatialSCE()) { 
+            geo::Vector_t offset = sce->GetCalPosOffsets(loc_atwires_alongpitch, plane.TPC);
```

Comment:
With the changes above, this may become:
```suggestion
            geo::Vector_t offset = sce->GetCalPosOffsets(loc_atwires_alongpitch, p.TPC);
```
(or `plane.ID().TPC`)